File: state/Desafio/Camila Aparecida_DesafioState/DesafioState/src/desafiostate/Eduardo Junior - DesafioState/src/MarioPequeno.java
public class MarioPequeno implements MarioState {

    @Override
    public MarioState pegarCogumelo() {
        System.out.println("Mario Grande");
        return new MarioGrande();
    }

    @Override
    public MarioState pegarFlor() {
        System.out.println("Mario Grande e Mario Fogo");
       return new MarioFogo();
    }

    @Override
    public MarioState pegarPena() {
        System.out.println("Mario Grande e Mario Capa");
        return new MarioCapa();
    }

    @Override
    public MarioState levarDano() {
        System.out.println("Mario Morto");
        return new MarioMorto();
    }  
}

File: PadraoState/PadraoState/Entitades/MarioPequeno.cs
using PadraoState.Interfaces;

namespace PadraoState.Entitades
{
    class MarioPequeno : MarioState
    {
        public MarioPequeno()
        {
            System.Console.WriteLine("Mario Pequeno");
        }        
        public MarioState PegarCogumelo()
        {
            return new MarioGrande();
        }

        public MarioState LevarDano()
        {
            return new MarioMorto();
        }

        public MarioState PegarFlor()
        {
            return new MarioFogo();
        }

        public MarioState PegarPena()
        {
            return new MarioPena();
        }
    }
}
